apps/dashboard/src/pages/edit-step-template-v2.tsx

```diff
@@ -18,24 +18,23 @@ export function EditStepTemplateV2Page() {
 
   const form = useForm({
     defaultValues,
+    values: step?.controls.values,
     shouldFocusError: false,
+    resetOptions: {
+      keepDirtyValues: true,
+    },
```

Comment:
values from the server are the source of truth but keeps dirty form fields; it's same as the `useEffect` lines that I removed